File: py/countPings.py
class RecentCounter:

    def __init__(self):
        self.pings = []

    def ping(self, t):
        self.pings.append(t)
        for p in self.pings[:]:
            if p < (t - 3000):
                self.pings.remove(p)

    


# Your RecentCounter object will be instantiated and called as such:
obj = RecentCounter()
obj.ping(1)
obj.ping(100)
obj.ping(3000)
obj.ping(3002)
obj.ping(3005)
obj.ping(7000)
obj.ping(7001)

File: js/countPings.js

class RecentCounter {
    constructor() {
        this.pings = [];
    }

    ping(t) {
        this.pings.push(t);
    
        const result = this.pings.filter((p) => {
            return p >= (t-3000)
        });
        
        this.pings = result;
        return this.pings.length;
    }

}

let rc = new RecentCounter();

rc.ping(1);
rc.ping(100);
rc.ping(3002);
rc.ping(7000);
rc.ping(8000);
rc.ping(13000);